coderd/users_test.go

```diff
@@ -1513,6 +1513,45 @@ func TestUsersFilter(t *testing.T) {
 		users = append(users, user)
 	}
 
+	// Add users with different creation dates for testing date filters
+	for i := 0; i < 3; i++ {
+		// nolint:gocritic
```

Comment:
Each of these `nolint` should mention that the system context is necessary here. (Also to make it clear which specific rule this is `nolint`ing)